recipes/cpc2/baseline/predict.py

```diff
@@ -0,0 +1,107 @@
+"""Make intelligibility predictions from HASPI scores."""
+import json
+import logging
+from pathlib import Path
+from typing import Optional
+
+import hydra
+import numpy as np
+import pandas as pd
+from omegaconf import DictConfig
+from scipy.optimize import curve_fit
+
+from clarity.utils.file_io import read_jsonl
+
+log = logging.getLogger(__name__)
+
+
+class LogisticModel:
+    """Class to represent a logistic mapping.
+
+    Fits a logistic mapping from input values x to output values y.
+    """
+
+    params: Optional[np.ndarray] = None  # The model params
+
+    def _logistic_mapping(self, x, x_0, k):
+        """
+        Logistic function
+            x_0 - x value of the logistic's midpoint
+            k - the logistic growth rate or steepness of the curve
+        """
+        return 100.0 / (1 + np.exp(-k * (x - x_0)))
+
+    def fit(self, x, y):
+        """Fit a mapping from x values to y values."""
+        initial_guess = [0.5, 1.0]  # Initial guess for parameter values
+        self.params, *_pcov = curve_fit(self._logistic_mapping, x, y, initial_guess)
+
+    def predict(self, x):
+        """Predict y values given x."""
+        # Note, fit() must be called before predictions can be made
+        assert self.params is not None
```

Comment:
Usually better to use a `try: ... except: ...` than `assert` I think as if `self.params` is `None` the following call `self._logistic_mapping()` would throw some sort of error, I think most probably a `NameError` as `self.params` won't have been defined.

Reply:
Agree. I've done this now by checking the params are not None and raising an exception as I think it communicate intended usage better than and try/except block